src/Logger.js

```diff
@@ -3,7 +3,25 @@ const readline = require('readline');
 const prettyError = require('./utils/prettyError');
 const emoji = require('./utils/emoji');
 
+const loggerMap = new WeakMap();
+
 class Logger {
+  static forAsset(asset) {
```

Comment:
Won't this bug the logger if multiple loggers exist?
Mainly like glitches with the status

Reply:
Oh yeah, this is just to quickly showcase the feature. A IPC system is way better like you did in the PR, like said in the description I'll remove this commit and use your PR once it gets merged.

However I'm not sure if Node.js scrambles the logs when `stdout` is shared across processes with `child_process` (it sure does when each `stdout` has it's own descriptor).

Reply:
Ow alright, thought this was another way to implement a singleton, guess i misunderstood.
About the workers stdout, I think it's best if we either don't use it or catch it on the master process and redirect it to the logger to make sure everything is streamlined to prevent console glitches

Reply:
I'm more for just using the IPC and not messing too much around `stdout` (Windows + `stdout` + multiple processes don't blend well in my experience).


`[bit off-topic]`
And I really like the IPC idea, all my plugins are needing an IPC at some point and I ended up making my own using UNIX sockets. If we could streamline this and make it accessible to plugin developers it could really help.
`[/bit off-topic]`

Reply:
Jasper's logger changes have been merged to master, so the IPC should just work now. You can just use the Logger singleton within the worker and things will be transparently proxied.